src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs

```diff
@@ -71,6 +74,7 @@ public override void Before(MethodInfo? methodUnderTest)
             // make sure we enable this for all unit tests
             AsynchronousOperationListenerProvider.Enable(enable: true, diagnostics: true);
             ExportProviderCache.SetEnabled_OnlyUseExportProviderAttributeCanCall(true);
+            FatalError.Handler = s_FatalErrorHandler;
```

Comment:
Looks like we are hitting the assertion in tests for setting the handler again
https://sourceroslyn.io/#Microsoft.CodeAnalysis/InternalUtilities/FatalError.cs,39